### ISSUE TITLE
About the config file of PTv3 on semantickitti

### ISSUE THREAD

**Issue by Vonct** [Pointcept/Pointcept#186]
Hi, I did not find the config file in the `configs/semantic_kitti` folder. Will it be released ?

**Comment by Gofinge** [comment]
Hi, releasing config for semantic_kitti is in our schedule. Typically I release a config after verifying the performance. Yet, recently, I have been fully devoted to a new research project, and  I will have some time to do this when this project goes into a stable training process. But I think it won't be difficult to reproduce the config when combine our config for PTv3-scenes and some other semantic_kitti configs.

**Comment by weilli** [comment]
> Hi, releasing config for semantic_kitti is in our schedule. Typically I release a config after verifying the performance. Yet, recently, I have been fully devoted to a new research project, and I will have some time to do this when this project goes into a stable training process. But I think it won't be difficult to reproduce the config when combine our config for PTv3-scenes and some other semantic_kitti configs.

Hi, is the semantic kitti PTv3 coming soon? I can't reproduce the accuracy of the paper, about 2% off. I'm so looking forward to it. Thanks.

**Comment by Gofinge** [comment]
So sorry for the long wait. Please give me some time, I will check this after the CVPR deadline. There would be an update of Pointcept after that.

**Comment by kabouzeid** [comment]
We would like to cite the SemanticKITTI numbers in our CVPR paper, however we also currently fail to reproduce the reported numbers by ~2 mIoU. Below is the config we're using right now. Copied from `semseg-spunet-v1m1-0-base`, then adjusted the learning rate and model to match the other PTv3 configs. Also removed PointClip aug, because it was even worse with it enabled.

Could you point us in the right direction? Perhaps a different batch size or different augmentations?

```python
_base_ = ["../_base_/default_runtime.py"]

# misc custom setting
batch_size = 8  # bs: total bs in all gpus
mix_prob = 0.8
empty_cache = False
enable_amp = True

# model settings
model = dict(
    type="DefaultSegmentorV2",
    num_classes=19,
    backbone_out_channels=64,
    backbone=dict(
        type="PT-v3m1",
        in_channels=4,
        order=["z", "z-trans", "hilbert", "hilbert-trans"],
        stride=(2, 2, 2, 2),
        enc_depths=(2, 2, 2, 6, 2),
        enc_channels=(32, 64, 128, 256, 512),
        enc_num_head=(2, 4, 8, 16, 32),
        enc_patch_size=(1024, 1024, 1024, 1024, 1024),
        dec_depths=(2, 2, 2, 2),
        dec_channels=(64, 64, 128, 256),
        dec_num_head=(4, 4, 8, 16),
        dec_patch_size=(1024, 1024, 1024, 1024),
        mlp_ratio=4,
        qkv_bias=True,
        qk_scale=None,
        attn_drop=0.0,
        proj_drop=0.0,
        drop_path=0.3,
        shuffle_orders=True,
        pre_norm=True,
        enable_rpe=False,
        enable_flash=True,
        upcast_attention=False,
        upcast_softmax=False,
        cls_mode=False,
        pdnorm_bn=False,
        pdnorm_ln=False,
        pdnorm_decouple=True,
        pdnorm_adaptive=False,
        pdnorm_affine=True,
        pdnorm_conditions=("nuScenes", "SemanticKITTI", "Waymo"),
    ),
    criteria=[
        dict(
            type="CrossEntropyLoss",
            weight=[
                3.1557,
                8.7029,
                7.8281,
                6.1354,
                6.3161,
                7.9937,
                8.9704,
                10.1922,
                1.6155,
                4.2187,
                1.9385,
                5.5455,
                2.0198,
                2.6261,
                1.3212,
                5.1102,
                2.5492,
                5.8585,
                7.3929,
            ],
            loss_weight=1.0,
            ignore_index=-1,
        ),
        dict(type="LovaszLoss", mode="multiclass", loss_weight=1.0, ignore_index=-1),
    ],
    # fmt: on
)

# scheduler settings
epoch = 50
eval_epoch = 50
optimizer = dict(type="AdamW", lr=0.002, weight_decay=0.005)
scheduler = dict(
    type="OneCycleLR",
    max_lr=[0.002, 0.0002],
    pct_start=0.04,
    anneal_strategy="cos",
    div_factor=10.0,
    final_div_factor=100.0,
)
param_dicts = [dict(keyword="block", lr=0.0002)]

# dataset settings
dataset_type = "SemanticKITTIDataset"
data_root = "data/semantic_kitti"
ignore_index = -1
names = [
    "car",
    "bicycle",
    "motorcycle",
    "truck",
    "other-vehicle",
    "person",
    "bicyclist",
    "motorcyclist",
    "road",
    "parking",
    "sidewalk",
    "other-ground",
    "building",
    "fence",
    "vegetation",
    "trunk",
    "terrain",
    "pole",
    "traffic-sign",
]

data = dict(
    num_classes=19,
    ignore_index=ignore_index,
    names=names,
    train=dict(
        type=dataset_type,
        split="train",
        data_root=data_root,
        transform=[
            # dict(type="RandomDropout", dropout_ratio=0.2, dropout_application_ratio=0.2),
            # dict(type="RandomRotateTargetAngle", angle=(1/2, 1, 3/2), center=[0, 0, 0], axis="z", p=0.75),
            dict(type="RandomRotate", angle=[-1, 1], axis="z", center=[0, 0, 0], p=0.5),
            # dict(type="RandomRotate", angle=[-1/6, 1/6], axis="x", p=0.5),
            # dict(type="RandomRotate", angle=[-1/6, 1/6], axis="y", p=0.5),
            dict(type="RandomScale", scale=[0.9, 1.1]),
            # dict(type="RandomShift", shift=[0.2, 0.2, 0.2]),
            dict(type="RandomFlip", p=0.5),
            dict(type="RandomJitter", sigma=0.005, clip=0.02),
            # dict(type="ElasticDistortion", distortion_params=[[0.2, 0.4], [0.8, 1.6]]),
            dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            # dict(type="PointClip", point_cloud_range=(-35.2, -35.2, -4, 35.2, 35.2, 2)),
            # dict(type="SphereCrop", sample_rate=0.8, mode="random"),
            # dict(type="SphereCrop", point_max=120000, mode="random"),
            # dict(type="CenterShift", apply_z=False),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
        ignore_index=ignore_index,
    ),
    val=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[
            dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="train",
                keys=("coord", "strength", "segment"),
                return_grid_coord=True,
            ),
            # dict(type="PointClip", point_cloud_range=(-35.2, -35.2, -4, 35.2, 35.2, 2)),
            dict(type="ToTensor"),
            dict(
                type="Collect",
                keys=("coord", "grid_coord", "segment"),
                feat_keys=("coord", "strength"),
            ),
        ],
        test_mode=False,
        ignore_index=ignore_index,
    ),
    test=dict(
        type=dataset_type,
        split="val",
        data_root=data_root,
        transform=[],
        test_mode=True,
        test_cfg=dict(
            voxelize=dict(
                type="GridSample",
                grid_size=0.05,
                hash_type="fnv",
                mode="test",
                return_grid_coord=True,
                keys=("coord", "strength"),
            ),
            crop=None,
            post_transform=[
                # dict(
                #     type="PointClip",
                #     point_cloud_range=(-35.2, -35.2, -4, 35.2, 35.2, 2),
                # ),
                dict(type="ToTensor"),
                dict(
                    type="Collect",
                    keys=("coord", "grid_coord", "index"),
                    feat_keys=("coord", "strength"),
                ),
            ],
            aug_transform=[
                [
                    dict(
                        type="RandomRotateTargetAngle",
                        angle=[0],
                        axis="z",
                        center=[0, 0, 0],
                        p=1,
                    )
                ],
                [
                    dict(
                        type="RandomRotateTargetAngle",
                        angle=[1 / 2],
                        axis="z",
                        center=[0, 0, 0],
                        p=1,
                    )
                ],
                [
                    dict(
                        type="RandomRotateTargetAngle",
                        angle=[1],
                        axis="z",
                        center=[0, 0, 0],
                        p=1,
                    )
                ],
                [
                    dict(
                        type="RandomRotateTargetAngle",
                        angle=[3 / 2],
                        axis="z",
                        center=[0, 0, 0],
                        p=1,
                    )
                ],
            ],
        ),
        ignore_index=ignore_index,
    ),
)

```

**Comment by vk7cp5ccvg-blip** [comment]
Hello, have you solved the issue with the configuration file for the KITTI dataset in PTV3? If you have, could you send me the code?

**Comment by Gofinge** [comment]
Thanks efforts from @Yujia-Zhang0913. The following code should be helpful: https://github.com/Pointcept/Pointcept/commit/bebf24bef0f3730b0c570fae080479bbbc8b68f9
https://github.com/Pointcept/Concerto/blob/main/demo/0_pca_outdoor.py

@vk7cp5ccvg-blip I prefer interacting with contributors using identifiable accounts, as it helps maintain constructive discussions. Please consider using your regular account for future issues.
